typed/parser.go

```diff
@@ -33,18 +33,19 @@ type YAMLParser interface {
 	FromYAML(object YAMLObject, typename string) (TypedValue, error)
 }
 
-type parser struct {
-	schema schema.Schema
+// Parser implements YAMLParser and allows introspecting the schema.
```

Comment:
You should probably drop the `YAMLParser` too then?

Reply:
It's OK to offer the interface (e.g. for testability). I can remove it though if you want.

Reply:
I created it because I wanted to hide the type more than because I wanted to fake it, so feel free to remove it

Reply:
ok one second.

Reply:
done